src/Psalm/Internal/Codebase/Properties.php

```diff
@@ -305,4 +316,18 @@ public function getPropertyType(
 
         return null;
     }
+
+    /**
+     * Returns ClassLikeStorage for the given class name, resolving aliases
+     */
+    private function getClasslikeStorage(string $fq_class_name): ?ClassLikeStorage
```

Comment:
Might it make sense to move this into ClassLikeStorageProvider like `getSafe` or such?

Reply:
The method uses `Classlikes::getUnAliasedName()` so if we move it the provider would have to depend on `Classlikes` which itself depends on `ClassLikeStorageProvider`. Such circular dependency would complicate initialization of both.

Reply:
Done